funnel/assets/js/utils/embedvideo.js

```diff
@@ -8,14 +8,22 @@ const Video = {
   */
   getVideoTypeAndId(url) {
     const regexMatch = url.match(
-      /(http:|https:|)\/\/(player.|www.)?(y2u\.be|vimeo\.com|youtu(be\.com|\.be|be\.googleapis\.com))\/(video\/|embed\/|live\/|watch\?v=|v\/)?([A-Za-z0-9._%-]*)(&\S+)?/
+      /(http:|https:|)\/\/(player.|www.)?(?<service>y2u\.be|vimeo\.com|youtu(be\.com|\.be|be\.googleapis\.com))\/(video\/|embed\/|live\/|watch\?v=|v\/)?(?<videoId>[A-Za-z0-9._%-]*)(&\S+)?((\?h=)(?<paramId>[^&]+))?/
     );
     let type = '';
     if (regexMatch && regexMatch.length > 5) {
-      if (regexMatch[3].indexOf('youtu') > -1 || regexMatch[3].indexOf('y2u') > -1) {
+      if (
+        regexMatch.groups.service.indexOf('youtu') > -1 ||
+        regexMatch.groups.service.indexOf('y2u') > -1
+      ) {
         type = 'youtube';
-      } else if (regexMatch[3].indexOf('vimeo') > -1) {
+      } else if (regexMatch.groups.service.indexOf('vimeo') > -1) {
         type = 'vimeo';
+        return {
+          type,
+          videoId: regexMatch.groups.videoId,
+          paramId: regexMatch.groups.paramId,
```

Comment:
This one return moved to the end of the condition would suffice. `paramId` will be null incase there is no match found.